frontend/app/components/api/api-v3/hal/hal-resource.service.ts

```diff
@@ -85,16 +86,22 @@ function halResource(halTransform, HalLink, $q) {
     }
 
     public $load() {
-      if (!this.$loaded) {
-        return this.$links.self().then(resource => {
-          this.$loaded = true;
+      if (this.$loaded) {
+        return $q.when(this);
+      }
 
-          angular.extend(this, resource);
-          return this;
-        });
+      if (!this.$loaded && this._loadingPromise) {
+        return this._loadingPromise;
       }
 
-      return $q.when(this);
+      this._loadingPromise = this.$links.self().then(resource => {
```

Comment:
Did you add the `loadingPromise` attribute because of error handling?
I am surely unaware of something here, but for now I don't see the difference between this code and the old one.